### PR TITLE
Sign using utf8 string in Metablock::new

### PR DESCRIPTION
**Pull request by danbev**
This commit updates the Metablock new function to use utf8 string when signing instead of the raw representation which is currently used.

The motivation for this change is that the verifying performed in `Metablock::verify` uses the utf8 string to generate the signature, and this can lead to a signature mismatch if the signatures are not created in the same manner.

This change also seems to be inline with the code in `MetablockBuilder::sign`, which also converts the raw bytes to utf8 before signing.

Signed-off-by: Daniel Bevenius <daniel.bevenius@gmail.com>

### CODE REVIEW

File: tests/runlib.rs [hunk 1]
```diff
@@ -131,6 +131,35 @@ fn in_toto_run_record_new_file() {
     dir.close().unwrap();
 }
 
+#[test]
+fn in_toto_run_new_line_in_stdout() {
```
Comment (adityasaky) [inline tests/runlib.rs:135]: Is this directly tied to the use of utf8? Just checking if I'm missing something.

**Review by adityasaky — APPROVED** [review]
LGTM.

File: tests/runlib.rs (hunk 1 above)
Comment (danbev) [reply tests/runlib.rs:135]: Sorry it is, but I named it very poorly :( 

This is trying to simulate how I ran into this issue and this test will fail without the other changes in this PR. I had a `clone-project` step which cloned a project, and I noticed that the signatures would be different when verifying which was the motivation for opening this PR.

**Comment by adityasaky** [comment]
Thanks, @danbev!

**Pull request merged**